src/main/java/hudson/plugins/promoted_builds/conditions/GroovyConditionDescriptor.java

```diff
@@ -4,41 +4,53 @@
 import hudson.PluginManager;
 import hudson.PluginWrapper;
 import hudson.model.AbstractProject;
+import hudson.model.Job;
+import hudson.model.TaskListener;
 import hudson.plugins.promoted_builds.PromotionConditionDescriptor;
+import javafx.concurrent.Task;
 import jenkins.model.Jenkins;
 
+import javax.annotation.Nonnull;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
  * Declared outside GroovyCondition as it depends on the optional 'script-security' dependency
  */
 @Extension(optional = true)
-public final class GroovyConditionDescriptor extends PromotionConditionDescriptor {
+public class GroovyConditionDescriptor extends PromotionConditionDescriptor {
 
     private static final Logger LOGGER = Logger.getLogger(GroovyConditionDescriptor.class.getName());
 
     public GroovyConditionDescriptor() {
         super(GroovyCondition.class);
     }
 
-    @Override
-    public boolean isApplicable(final AbstractProject<?, ?> item) {
-        // TODO switch to Jenkins.getActiveInstance() once bumped to 1.590
-        final Jenkins jenkins = Jenkins.getInstance();
-        if (jenkins == null) {
-            // Jenkins not started or shut down
-            return false;
-        }
-        final PluginManager pluginManager = jenkins.getPluginManager();
-        if (pluginManager == null) {
-            LOGGER.log(Level.WARNING, "No PluginManager");
-            return false;
+   public boolean isApplicable(@Nonnull Job<?,?> item, TaskListener listener){
+        if(item instanceof AbstractProject){
+            return isApplicable((AbstractProject)item, TaskListener.NULL);
+        }else{
```

Comment:
minor nitpick: formatting. Leave blank spaces  for if and else statements like everywhere else.

Reply:
Your IDE could be configured to do it for you